UnityProject/Assets/Scripts/UI/Objects/Research/ResearchServer/AvailableTechEntry.cs

```diff
@@ -0,0 +1,59 @@
+using UI.Core.Net.Elements;
+using UI.Core.NetUI;
+using UnityEngine;
+using Systems.Research.Data;
+using Systems.Research;
+
+namespace UI.Objects.Research
+{
+	public class AvailableTechEntry : DynamicEntry
+	{
+		[SerializeField] private NetText_label techDescription;
+		[SerializeField] private NetText_label techName;
+		[SerializeField] private NetText_label techPrice;
+		[SerializeField] private EmptyItemList spriteList;
+		
+
+		private Technology technologyToUnlock;
+		private Techweb techWeb;
+		private CustomNetworkManager networkManager;
+
+
+		public void Initialise(Technology technology, Techweb techWeb)
+		{
+			networkManager = CustomNetworkManager.Instance;
+			technologyToUnlock = technology;
+			this.techWeb = techWeb;
+
+			techName.MasterSetValue(technology.DisplayName);
+			techDescription.MasterSetValue(technology.Description);
+			techPrice.MasterSetValue(technology.ResearchCosts.ToString());
+			SetSprites();
+		}
+
+		private void SetSprites()
+		{
+			int unlockCount = technologyToUnlock.DesignIDs.Count;
+			spriteList.SetItems(unlockCount);
+		
+			for(int i = 0; i < unlockCount; i++)
```

Comment:
why a for loop isntead of a foreach loop?

Reply:
Responded to this comment elsewhere, its the same reason for all the for loops in this PR

Reply:
alright carry on